python/torch_mlir_e2e_test/test_suite/basic.py

```diff
@@ -1420,6 +1420,26 @@ def BroadcastListConstructWithMinusOneModule_basic(module, tu: TestUtils):
 
 # ==============================================================================
 
+class BroadcastDifferentRanktWithMinusOneModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([3, 1, 8], torch.float32, True),
+    ])
+    def forward(self, x):
+        return torch.broadcast_to(x, [10, -1, -1, -1])
```

Comment:
What would be the tosa code generated here? I'm thinking that this should be a `reshape` + `add`, but I didn't see a reshape being added in the lowering.

Reply:
The reshape is generated by `TosaMakeBroadcastable` pass that runs later in the pipeline. Essentially there is no need for us to introduce reshapes since the main purpose of that pass is to introduce reshapes and enable broadcasting (`llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaMakeBroadcastable.cpp:94` calls function `reshapeLowerToHigher`). The IR for this example is:
`module attributes {torch.debug_module_name = "BroadcastDifferentRanktWithMinusOneModule"} {
  func.func @forward(%arg0: tensor<3x1x8xf32>) -> tensor<10x3x1x8xf32> {
    %0 = "tosa.const"() <{value = dense<0.000000e+00> : tensor<10x3x1x8xf32>}> : () -> tensor<10x3x1x8xf32>
    %1 = "tosa.reshape"(%arg0) <{new_shape = array<i64: 1, 3, 1, 8>}> : (tensor<3x1x8xf32>) -> tensor<1x3x1x8xf32>
    %2 = "tosa.add"(%1, %0) : (tensor<1x3x1x8xf32>, tensor<10x3x1x8xf32>) -> tensor<10x3x1x8xf32>
    return %2 : tensor<10x3x1x8xf32>
  }
}`

Reply:
BTW, we also pass one more test in `make_fx_tosa` =)